### PR TITLE
SISRP-16879 SISRP-16880 Adapt My Academics and My Classes to incorporate EdoOracle

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.berkeley.edu/browse/SISRP-16879
https://jira.berkeley.edu/browse/SISRP-16880

### CODE REVIEW

**Comment by johncrossman** [comment]
@paulkerschen This looks good to me but we need @redconfetti or @raydavis to review, too.

File: app/models/my_academics/semesters.rb [hunk 1]
```diff
@@ -9,8 +9,14 @@ def initialize(uid)
     def merge(data)
       @filtered = data[:filteredForDelegate]
 
-      all_user_courses = CampusOracle::UserCourses::All.new(user_id: @uid)
-      enrollments = @filtered ? all_user_courses.get_enrollments_summary : all_user_courses.get_all_campus_courses
+      legacy_user_courses = CampusOracle::UserCourses::All.new(user_id: @uid)
+      edo_user_courses = EdoOracle::UserCourses::All.new(user_id: @uid)
+      if @filtered
+        enrollments = legacy_user_courses.get_enrollments_summary.merge(edo_user_courses.get_enrollments_summary)
```
Comment (redconfetti) [inline app/models/my_academics/semesters.rb]: @paulkerschen - This merge may not honor the proper sorting

**Comment by redconfetti** [comment]
MyAcademics::Teaching needs to have an update for EdoOracle switching

**Comment by paulkerschen** [comment]
Amendment pushed following on yesterday's review with @redconfetti:
- Better guarantees for correct semester ordering when pulling from multiple sources;
- Source-switching logic added to MyAcademics::Teaching#courses_list_from_ccns.

File: spec/models/my_academics/semesters_spec.rb [hunk 2]
```diff
@@ -1,15 +1,12 @@
 describe MyAcademics::Semesters do
 
-  let(:initial_feed) { {} }
-  let(:feed) { feed = initial_feed; MyAcademics::Semesters.new(random_id).merge(feed); feed }
+  let(:feed) { {}.tap { |feed| MyAcademics::Semesters.new(random_id).merge(feed) } }
```
Comment (redconfetti) [inline spec/models/my_academics/semesters_spec.rb]: @paulkerschen nice!

**Pull request merged**